src/java.desktop/share/classes/javax/swing/text/html/HTMLDocument.java

```diff
@@ -2708,15 +2708,15 @@ private void adjustEndElement() {
         }
 
         private Element[] getPathTo(int offset) {
-            Stack<Element> elements = new Stack<Element>();
+            ArrayDeque<Element> elements = new ArrayDeque<Element>();
             Element e = getDefaultRootElement();
             int index;
             while (!e.isLeaf()) {
                 elements.push(e);
                 e = e.getElement(e.getElementIndex(offset));
             }
             Element[] retValue = new Element[elements.size()];
-            elements.copyInto(retValue);
+            elements.toArray(retValue);
```

Comment:
Same comment as in some other review, the copyInto and toArray works differently, it is better to use return elements.toArray(new Element[0]);

Reply:
Updated